### ISSUE TITLE
Non availability of `PBS_*` environment variables on compute nodes in the reservation

### ISSUE THREAD

**Issue by mahendrapaipuri** [openpbs/openpbs#2472]
Hello,

I am doing some tests with PBSPro local cluster that I deployed with Docker compose. I am using the recipes from this daskqueue [repository](https://github.com/dask/dask-jobqueue/tree/main/ci/pbs). It works very well for my testing. However, I have a question regarding the environment variables of batch job. 

With this Docker compose, I spin up a master node and two slave nodes. I submit the jobs from the master node to the slaves. When I check for PBS related environment variables (`PBS_*`) they are only available on the first slave node and are not available on the second. Is it a normal behaviour? If so, how can I access all `PBS_*` environment variables on all the nodes in the reservation? I do not have access to production machines that uses PBS workload scheduler, so I can not verify this behaviour elsewhere.

This is not really an issue, but more of a help! I tried to look online for any leads but ended up empty handed. 

Thanks!!

**Comment by vstumpf** [comment]
How are you accessing the second node? I assume you're using an interactive job to get onto the first node. The execution daemon (the pbs mom) sets up your environment in the shell you're given as an interactive job. If you ssh to the second node, the mom can't do this, you lose the "job context", so none of the environment variables are set.

We have some commands that let you run things on other machines within the job. These are pbsdsh and pbs_tmrsh. OpenMPI, when built correctly, can use the `tm` interface in `libpbs` to spawn tasks on the other nodes in the job.

**Comment by mahendrapaipuri** [comment]
@vstumpf Thanks a lot for the reply. Well, actually I am launching a python script on both nodes using native `mpirun` and inside the python script I use the scheduler related env variables, which in the current case variables `PBS_*`. On the slave node 1, these env variables are found in the script whereas on the other node they were not present. 

But I solved the issue with your tip! In fact, I installed the OpenMPI using RPM packages and it did not have support for PBS. I compiled OpenMPI from sources using `./configure --with-tm --enable-mpi-interface-warning --enable-shared --enable-static --enable-cxx-exceptions` and everything worked as you said. Now, I could access the PBS related env variables on all nodes.

Sorry for the naive question. As I told I am quite new to PBS and I thought support for PBS is there by default in OpenMPI as it does for SLURM. I am closing this issue now! Thanks again!!
